internal/generate/olm-catalog/descriptor/descriptor.go

```diff
@@ -51,19 +49,55 @@ func GetCRDDescriptionForGVK(apisDir string, gvk schema.GroupVersionKind) (olmap
 	if strings.Contains(group, ".") {
 		group = strings.Split(group, ".")[0]
 	}
-	apiDir := filepath.Join(apisDir, group, gvk.Version)
-	universe, err := getTypesFromDir(apiDir)
+
+	// Check if apisDir exists
+	exists, err := isDirExist(apisDir)
 	if err != nil {
-		if os.IsNotExist(err) {
-			return olmapiv1alpha1.CRDDescription{}, ErrAPIDirNotExist
-		}
 		return olmapiv1alpha1.CRDDescription{}, err
 	}
-	apiPkg := path.Join(projutil.GetGoPkg(), filepath.ToSlash(apiDir))
-	pkgTypes, err := getTypesForPkg(apiPkg, universe)
+	if !exists {
+		log.Debugf("Could not find API types directory: %s", apisDir)
+		return olmapiv1alpha1.CRDDescription{}, ErrAPIDirNotExist
+	}
+
+	// Check if the kind pkg is at the expected layout
```

Comment:
Is there a reason to check for different layouts? Given `apisDir` is configurable, `getTypesFromDirRecursive()` will find types regardless right?

Reply:
So `apisDir` is only configurable to be the API types root dir from where we construct an exact path to the types subdirectory for a GVK e.g `<api-dir>/<group>/<version>` or `<api-dir>/<version>`.

For muli-group layouts e.g `pkg/apis/cache/v1alpha1` and `pkg/apis/app/v1alpha1` if we just parse `pkg/apis` recursively and then try to find the types for the GVK `app/v1alpha1 App` from the universe of pkgs we'll have a collision when looking for a pkg named `v1alpha1`.

So that's why we first detect the exact layout for multi or single group pkgs and only parse the exact pkg dir and not the whole of `pkg/apis`.

Also I've realized `getTypesFromDirRecursive()` is named incorrectly since it returns the universe of pkgs [`type Universe map[string]*Package`](https://github.com/kubernetes/gengo/blob/master/types/types.go#L224) from the dir. The exact pkg types `[]*types.Type` are then filtered out by `getTypesForPkgPath`.

Going to rename `getTypesFromDirRecursive()` to `getPkgsFromDirRecursive()`